tlsn/tlsn-prover/src/tls/prove.rs

```diff
@@ -169,38 +146,28 @@ impl Prover<ProveState> {
             handshake_decommitment,
         };
 
-        let mut finalize_fut = Box::pin(async move {
-            let mut channel = mux_ctrl.get_channel("finalize").await?;
-
-            _ = vm
-                .finalize()
-                .await
-                .map_err(|e| ProverError::MpcError(Box::new(e)))?
-                .expect("encoder seed returned");
-
-            // This is a temporary approach until a maliciously secure share conversion protocol is implemented.
-            // The prover is essentially revealing the TLS MAC key. In some exotic scenarios this allows a malicious
-            // TLS verifier to modify the prover's request.
-            gf2.reveal()
-                .await
-                .map_err(|e| ProverError::MpcError(Box::new(e)))?;
-
-            // Send session_info to the verifier
-            channel.send(TlsnMessage::SessionInfo(session_info)).await?;
-
-            Ok::<_, ProverError>(())
-        })
-        .fuse();
-
-        futures::select_biased! {
-            res = finalize_fut => res?,
-            _ = ot_fut => return Err(OTShutdownError)?,
-            _ = &mut mux_fut => return Err(std::io::Error::from(std::io::ErrorKind::UnexpectedEof))?,
-        };
+        mux_fut
+            .poll_with(async move {
+                ot_recv.accept_reveal(&mut ctx).await?;
+
+                _ = vm
+                    .finalize()
+                    .await
+                    .map_err(|e| ProverError::MpcError(Box::new(e)))?
+                    .expect("encoder seed returned");
+
+                // Send session_info to the verifier
+                io.send(session_info).await?;
+
+                Ok::<_, ProverError>(())
+            })
+            .await?;
+
+        if !mux_fut.is_complete() {
+            mux_ctrl.mux().close();
+            mux_fut.await?;
```

Comment:
need to add a comment

// Wait for the verifier to correctly close the connection.